internal/engine/pod.go

```diff
@@ -122,16 +123,15 @@ func trackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*store.
 		(isAncestorMatch && runtime.PodAncestorUID != ancestorUID) {
 		runtime.PodAncestorUID = ancestorUID
 		runtime.Pods = make(map[k8s.PodID]*store.Pod)
-		pod := &store.Pod{
+		podInfo := &store.Pod{
 			PodID:      podID,
 			StartedAt:  startedAt,
 			Status:     status,
 			Namespace:  ns,
 			HasSynclet: hasSynclet,
 		}
-		runtime.Pods[podID] = pod
-		ms.RuntimeState = runtime
-		return pod, true
+		maybeAttachPodToManifest(ms, runtime, podInfo, currentDeploy)
```

Comment:
the `maybe` here doesn't seem correct. We've already started mutating the state of the runtime a couple lines above based on this new pod info. Why would we want to start mutating the state, then abandon mid-stream?

in general, it would be better if the currentDeploy check was better unified with the other `if` checks.